### ISSUE TITLE
detect when people get wrong to extend BaseBuild or BuildBuild

### ISSUE THREAD

**Issue by cvogt** [cvogt/cbt#250]
Seems to easily happen, probably due to copy and paste

**Comment by tim-zh** [comment]
Better naming should be enough to solve the problem. Maybe MetaBuild instead of BuildBuild?

**Comment by cvogt** [comment]
@tim-zh I agree that better naming would probably help. Not convinced of calling it MetaBuild yet. This may be completely subjective, but meta seems more ambiguous and generally a bit overused. There is also Scala meta and other things. It's appropriate, but it seems less intuitive to me, than BuildBuild. Meaning if you see the name for the first time, how easily can you predict it's role or meaning.

This may be solved by another change I have in mind: https://github.com/cvogt/cbt/issues/270 . This would have ordinary builds extend `ScalaBuild` and meta builds extend `BuildBuild`.

WDYT?

**Comment by tim-zh** [comment]
I remember the first time I saw `BuildBuild`, it was absolutely confusing. Renaming `BaseBuild` will help to distinguish it from `BuildBuild`, but won't solve the problem with unobvious name.
In my mind, `MetaBuild` fits perfectly as it denotes "build for builds". Alternatively, it could be just `BuildForBuild` :)

**Comment by cvogt** [comment]
So MetaBuild would have been more self-explanatory than BuildBuild?

**Comment by cvogt** [comment]
one more vote for MetaBuild: https://gitter.im/cvogt/cbt?at=57f7f9820aa72e3c5be2b54e

**Comment by cvogt** [comment]
Related to #174
